contracts/RevisedPixelmonTrainerAdventure/PxChainlinkManager.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.16;
+
+/// @title Pixelmon Trainer Adventure Smart Contract
+/// @author LiquidX
+/// @notice This smart contract provides configuration for the Trainer Adventure event on Pixelmon
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+error NotAllowedToCall();
+
+contract PxChainlinkManager is EIP712, Ownable, VRFConsumerBaseV2 {
+    /// @dev Signing domain for the purpose of creating signature
+    string public constant SIGNING_DOMAIN = "Pixelmon-Trainer-Adventure";
+    /// @dev signature version for creating and verifying signature
+    string public constant SIGNATURE_VERSION = "1";
+
+    /// @dev Signer wallet address for signature verification
+    address public SIGNER;
+
+    /// @notice Amount of random number requested to Chainlink
+    uint32 public constant Random_Number_Count = 3;
+
+    /// @notice Struct object to send request to Chainlink
+    /// @param fulfilled Whether the random words has been set or not
+    /// @param exists Whether the request has been sent or not
+    /// @param weekNumber Draw week number
+    /// @param randomWords Random words from Chainlink
+    struct Request {
+        bool fulfilled;
+        bool exists;
+        uint256 weekNumber;
+        uint256[] randomWords;
+    }
+
+    /// @notice Struct object to save weekly chainlink random numbers 
+    /// @param randomNumbers weekly chainlink random numbers
+    struct WeeklyRandomNumber {
+        uint256[] randomNumbers;
+    }
+
+    /// @notice Map of weekly chainlink random numbers
+    mapping(uint256 => WeeklyRandomNumber) weeklyRandomNumbers;
+
+    /// @notice address of trainer adventure smart contract
+    address public trainerAdventureContractAddress;
+
+    // @notice The maximum gas price to pay for a request to Chainlink in wei.
```

Comment:
why does keyhash's comment says gas price?

Reply:
Chainlink library takes 3 different hash as input for three different maximum gas price

Reply:
keyHash The gas lane to use, which specifies the maximum gas price to bump to.
More https://docs.chain.link/docs/vrf/v2/subscription/supported-networks/#configurations